baseapp/test_helpers.go

```diff
@@ -15,10 +15,11 @@ func (app *BaseApp) Check(txEncoder sdk.TxEncoder, tx sdk.Tx) (sdk.GasInfo, *sdk
 	if err != nil {
 		return sdk.GasInfo{}, nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "%s", err)
 	}
-	return app.runTx(runTxModeCheck, bz)
+	gasInfo, result, err, _ := app.runTx(runTxModeCheck, bz)
+	return gasInfo, result, err
 }
 
-func (app *BaseApp) Simulate(txBytes []byte) (sdk.GasInfo, *sdk.Result, error) {
+func (app *BaseApp) Simulate(txBytes []byte) (sdk.GasInfo, *sdk.Result, error, sdk.Context) {
```

Comment:
#nit I think it is standard go formatting practice to put error at the end of a list of return params.

Reply:
Good catch.  Thanks!